### PR TITLE
Bugfix BC default

### PR DESCRIPTION
**Pull request by simone-silvestri**
for `LatitudeLongitudeGrid`s spanning -90:90 with a `Nothing` location in the latitude direction.

For reference on main:
```julia
julia> grid = LatitudeLongitudeGrid(size = (600, 100, 1), latitude = (-90, 90), longitude = (0, 360), z = (0, 1))
600×100×1 LatitudeLongitudeGrid{Float64, Periodic, Bounded, Bounded} on CPU with 3×3×1 halo and with precomputed metrics
├── longitude: Periodic λ ∈ [0.0, 360.0)  regularly spaced with Δλ=0.6
├── latitude:  Bounded  φ ∈ [-90.0, 90.0] regularly spaced with Δφ=1.8
└── z:         Bounded  z ∈ [0.0, 1.0]    regularly spaced with Δz=1.0

julia> minimum_xspacing(grid)
ERROR: ArgumentError: Cannot specify north boundary condition ValueBoundaryCondition: Oceananigans.BoundaryConditions.PolarValue{OffsetArrays.OffsetArray{Float64, 3, Array{Float64, 3}}, Symbol} on a field at nothing!
Stacktrace:
  [1] validate_boundary_condition_location(bc::BoundaryCondition{…}, loc::Nothing, side::Symbol)
    @ Oceananigans.Fields ~/development/Oceananigans.jl/src/Fields/field.jl:57
  [2] validate_boundary_conditions(loc::Tuple{…}, grid::LatitudeLongitudeGrid{…}, bcs::FieldBoundaryConditions{…})
    @ Oceananigans.Fields ~/development/Oceananigans.jl/src/Fields/field.jl:76
  [3] #apply_regionally!#58
    @ ~/development/Oceananigans.jl/src/Utils/multi_region_transformation.jl:121 [inlined]
  [4] apply_regionally!
    @ ~/development/Oceananigans.jl/src/Utils/multi_region_transformation.jl:118 [inlined]
  [5] macro expansion
    @ ~/development/Oceananigans.jl/src/Utils/multi_region_transformation.jl:206 [inlined]
  [6] Field(loc::Tuple{…}, grid::LatitudeLongitudeGrid{…}, data::OffsetArrays.OffsetArray{…}, bcs::FieldBoundaryConditions{…}, indices::Tuple{…}, op::Nothing, status::Nothing)
    @ Oceananigans.Fields ~/development/Oceananigans.jl/src/Fields/field.jl:101
  [7] Field(loc::Tuple{…}, grid::LatitudeLongitudeGrid{…}, T::DataType; indices::Tuple{…}, data::OffsetArrays.OffsetArray{…}, boundary_conditions::FieldBoundaryConditions{…}, operand::Nothing, status::Nothing)
    @ Oceananigans.Fields ~/development/Oceananigans.jl/src/Fields/field.jl:194
  [8] minimum(f::Function, c::KernelFunctionOperation{…}; condition::Nothing, mask::Float64, dims::Function)
    @ Oceananigans.Fields ~/development/Oceananigans.jl/src/Fields/field.jl:710
  [9] minimum
    @ ~/development/Oceananigans.jl/src/Fields/field.jl:701 [inlined]
 [10] minimum
    @ ~/development/Oceananigans.jl/src/Fields/field.jl:721 [inlined]
 [11] minimum_xspacing(grid::LatitudeLongitudeGrid{…})
    @ Oceananigans.Grids ~/development/Oceananigans.jl/src/Grids/nodes_and_spacings.jl:160
 [12] top-level scope
    @ REPL[17]:1
Some type information was truncated. Use `show(err)` to see complete types.

julia>
```

### CODE REVIEW

**Comment by navidcy** [comment]
Was this bug always there or appeared after a recent Pr?

**Comment by simone-silvestri** [comment]
> I’m confused. The grid has nothing on location? The Field I presume?

yeah, the default BC depends on the grid type but also on the location of the field (for example, by default fields on `Face` have an impenetrable boundary condition)

> Was this bug always there or appeared after a recent PR?

It's just a product of a refactor. I have added some simple tests.

File: src/BoundaryConditions/field_boundary_conditions.jl [hunk 1]
```diff
@@ -277,26 +277,42 @@ regularize_south_boundary_condition(bc::DefaultBoundaryCondition, grid::Latitude
...
     φnorth = @allowscalar φnode(grid.Ny+1, grid, Face()) 
     cca_loc = loc[1] == Center && loc[2] == Center # scalar
     default_bc = default_prognostic_bc(topology(grid, 2)(), loc[2](), default)
     return φnorth ≈ 90 && cca_loa ? PolarBoundaryCondition(grid, :north, loc[3]) : default_bc
 end
 
 function default_prognostic_bc(grid::LatitudeLongitudeGrid, ::Val{:south}, loc, default)
+    if loc[2] == Nothing
+        return nothing
+    end
```
Comment (glwagner) [inline src/BoundaryConditions/field_boundary_conditions.jl:293]: should the function signature read LX, LY, LZ rather than `loc`?

**Comment by glwagner** [comment]
It's good to fix this bug, but in my opinion the code design here may warrant a bit more of a refactor than is being proposed.

**Review by glwagner — APPROVED** [review]
ok, I like how the code reads now, its clear
